L/libzenohc/build_tarballs.jl

```diff
@@ -0,0 +1,42 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "libzenohc"
+version = v"0.10.1"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/eclipse-zenoh/zenoh-c.git", "a0bacbb65b11e1e5ecedb3b3888beba78bf227cd")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/zenoh-c
+cmake -B build -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DZENOHC_CARGO_CHANNEL=$RUSTUP_TOOLCHAIN -DZENOHC_CUSTOM_TARGET=$CARGO_BUILD_TARGET
+cmake --build build --parallel ${nproc}
+cmake --install build
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("x86_64", "macos"; ),
+    Platform("aarch64", "macos"; ),
+    Platform("x86_64", "freebsd"; )
+]
+
```

Comment:
What's the problem with the other platforms?

Reply:
They did not compile successfully with the given cmake commands. Unfortunately, I don't have the capacity to follow through and check / fix the build for every platform. Taking i686-linux-gnu as an example, apparently there is a compilation problem with a dependency (see https://github.com/briansmith/ring/issues/1795).

Is supporting all platforms a requirement for merging into main?

Reply:
No, it isn't mandatory, but I was curious because for the most part we should be able to build Rust for all platforms (except for 32-bit Windows)